src/prefect/engine/results/__init__.py

```diff
@@ -1,3 +1,36 @@
+"""
+`Result` subclasses are the backbone of tracking the value, type and optional persistence method of return values from tasks by exposing a `read` / `write` / `exists` interface for common storage backends. Results can also be instantiated directly by the user in a task to use those methods to interact with persistent storage and track data besides a task's return value.
+
+A results `read` / `write` / `exists` methods depends on a result's `location` attribute, which can be a concrete string or a templated string that will be formatted at time of `write` using `prefect.context`.
+
+Note that a result's `read` and `write` methods return new `Result` instances, the former with their `location` attribute formatted, and the latter with the `value` attribute hydrated.
+
+For example, here is how you would use a result in a task directly to read and write arbitrary pieces of data:
+
+```python
+from prefect import task
+from prefect.engine.results import S3Result
+
+MY_RESULTS = S3Result(bucket='my_bucket')
+
+@task
+def my_example_task():
+    # read data from a file in the bucket
+    my_task_data = MY_RESULTS.read(location="some_data_in_my_bucket.csv")
+    print(my_task_data.value) # is the deserialized data that was in the file s3://my_bucket/some_data_in_my_bucket.csv
+    
+    # write data to a templated filename in the bucket
+    data = 3
+    my_task_data = MY_RESULTS.write(data, location="{task_name}.txt")
```

Comment:
I think you need to explicitly provide `**kwargs` to `write` in order for any templating to take place

Reply:
Fixed in b26d2c1799347e2a4bf9efddaca0bf04f0fceac5